src/Evolution/Systems/ScalarWave/Tags.hpp

```diff
@@ -104,5 +105,33 @@ struct EvolvedFieldsFromCharacteristicFields : db::SimpleTag {
     return "EvolvedFieldsFromCharacteristicFields";
   }
 };
+
+/// The energy density of the scalar wave system
+template <size_t SpatialDim>
+struct EnergyDensity : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+/// @{
+/*!
+ * \brief Computes the energy density of the scalar wave system.
+ *
+ * Below is the fucntion used to calculate the energy density.
+ * \f{align*}
+ * \epsilon = \frac{1}{2}\left( \Pi^{2} + \abs{\Phi}^{2} \right)
+ * \f}
+ */
+template <size_t SpatialDim>
+struct EnergyDensityCompute : EnergyDensity<SpatialDim>, db::ComputeTag {
+  using base = EnergyDensity<SpatialDim>;
+  using return_type = Scalar<DataVector>;
+  static void function(
```

Comment:
you'll need to do a `static_cast` and turn this into something similar to https://github.com/sxs-collaboration/spectre/blob/08973d05a7262714ec71b006b40fc88b4f1c9058/src/PointwiseFunctions/GeneralRelativity/Lapse.hpp#L66